draft-ietf-dnsop-svcb-https.md

```diff
@@ -959,7 +963,7 @@ approach such as Happy Eyeballs {{!HappyEyeballsV2}}.
 When only "ipv4hint" is present, IPv6-only clients may synthesize
 IPv6 addresses as specified in {{!RFC7050}} or ignore the "ipv4hint" key and
 wait for AAAA resolution ({{client-behavior}}).  Recursive resolvers MUST NOT
-perform DNS64 ({{!RFC6147}}) on parameters within a SVCB record.
+perform DNS64 ({{?RFC6147}}) on parameters within a SVCB record.
```

Comment:
It feels odd to use an informational reference in specifying a normative requirement.  In saying you MUST NOT take some action, we inherently make knowing what not to do necessary to complying.  (In other words, I disagree with @kaduk here, and he accepts my disagreement.)  Vote of the authors?

Reply:
My intuition was similar to @MikeBishop's, but I view this as a process question on which @kaduk may be authoritative.

Reply:
Would "SHOULD NOT" be better, with the warning that doing so would break any DNSSEC signatures? 


In General Requirements we say:

> Recursive resolvers MUST be able to convey SVCB records with unrecognized SvcParamKeys, and MAY treat the entire SvcParams portion of the record as opaque, even if the contents are invalid. Alternatively, recursive resolvers MAY report an error such as SERVFAIL to avoid returning a SvcParamValue that is invalid according to the SvcParam's specification. For complex value types whose interpretation might differ between implementations or have additional future allowed values added (e.g. URIs or "alpn"), resolvers SHOULD limit validation to specified constraints.

but we don't say anything about modifying parameters.

Reply:
I don't think MUST vs. SHOULD is relevant to the question of whether the reference is normative.

I do think this is a MUST.  Violations of this rule will result in compatibility failures.

Reply:
```suggestion
perform DNS64 ({{?RFC6147}}) on parameters within a SVCB record
as this would result in DNSSEC signature and other compatibility failures.
```

?  (and leave as a MUST?)
But this is odd in-general as it is saying "don't get creative here and try
and extrapolate from something informational!"

Reply:
I'm fine with this text change, but it sounds like it doesn't quite capture your intent.  Maybe the solution is to drop the normative language and say something like

> Note that any DNS64 AAAA record synthesis ({{Section 5.1.7 of ?RFC6147}}) applies only to AAAA queries, and does not extend to IP hints conveyed in SVCB records.

Reply:
> I'm fine with this text change, but it sounds like it doesn't quite capture your intent. Maybe the solution is to drop the normative language and say something like
> 
> > Note that any DNS64 AAAA record synthesis ({{Section 5.1.7 of ?RFC6147}}) applies only to AAAA queries, and does not extend to IP hints conveyed in SVCB records.

This seems like a pretty good approach to me.  The specific wording might get tweaked further (the bit around "applies only to" catches my eye as something I might want to edit, but I can't put my finger on exactly why at the moment), but it seems okay to frame this statement as a factual clarification for the scope of RFC 6147 rather than a new normative statement with normative keywords.

Reply:
Oops, but there were more comments elsewhere that render this proposal obsolete anyway.  I should have read on before commenting, sorry!